tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -4210,25 +4210,36 @@ void VerifySampleResults(const UINT *pPixels, UINT width) {
   UINT ylod = 0;
   // Each pixel contains 4 samples and 4 LOD calculations.
   // 2 of these (called 'left' and 'right') have X values that vary and a
-  // constant Y 2 others (called 'top' and 'bot') have Y values that vary and a
-  // constant X Only of the X variant sample results and one of the Y variant
-  // results are actually reported for the pixel. The other 2 serve as "helpers"
-  // to the other pixels in the quad. On the left side of the quad, the 'left'
-  // samples are reported. Op the top of the quad, the 'top' samples are
-  // reported and so on. The varying coordinate values alternate between zero
-  // and a value whose magnitude increases with the index. As a result, the LOD
-  // level should steadily increas. Due to vagaries of implementation, the same
-  // derivatives in both directions might result in different levels for
+  // constant Y. 2 others (called 'top' and 'bot') have Y values that vary and a
+  // constant X. Only one of the X variant sample results and one of the Y
+  // variant results are actually reported for the pixel. The other 2 serve as
+  // "helpers" to the other pixels in the quad. On the left side of the quad,
+  // the 'left' samples are reported. On the top of the quad, the 'top' samples
+  // are reported and so on. The varying coordinate values alternate between
+  // zero and a value whose magnitude increases with the index. As a result, the
+  // LOD level should steadily increase. Due to vagaries of implementation, the
+  // same derivatives in both directions might result in different levels for
   // different locations in the quad. So only comparisons between sample results
   // and LOD calculations and ensuring that the LOD increased and reaches the
   // max can be tested reliably.
+
+  // The results are stored in quad z-order (top-left (#0), top-right (#1),
+  // bottom-left (#2), bottom-right (#3)) and need to be evaluated as such.
+  // The X-derivative-LOD should not decrease when going from quad pixel #0->#1
+  // and #2->#3 or #3->#0, but not for #1->#2 where is is absolutely valid for
+  // the X-derivative-LOD to decrease, because pixel #2's X coordinate is less
+  // than pixel #1's. Therefore the test skips verification of X-derivative-LOD
+  // on quad pixel #2.
```

Comment:
The "not" on line 4229 is negating the earlier "not decrease", which is a bit confusing. I would make a new sentence that states outright that the `#1->#2` transition should involve a decrease.
  
There are simpler ways to update the text. I got a bit carried away.

```suggestion
  // The X-derivative-LOD should not decrease when going from quad pixel #0->#1,
  // #2->#3, or #3->#0. For #1->#2 the end of the typewriter "line" is reached and
  // zags left resulting in a smaller x value. So, it is absolutely valid for the 
  // X-derivative-LOD to decrease. Therefore, the test skips verification of X-derivative-LOD
  // on quad pixel #2.
```